mlflow/sklearn/utils.py

```diff
@@ -286,61 +286,57 @@ def _get_classifier_artifacts(fitted_estimator, fit_args, fit_kwargs):
     """
     import sklearn
 
+    if not _is_plotting_supported():
+        return []
+
     fit_arg_names = _get_arg_names(fitted_estimator.fit)
-    X, y_true, y_pred = _get_samples_labels_and_predictions(
-        fitted_estimator, fit_args, fit_kwargs, fit_arg_names
-    )
+    X, y_true = _get_Xy(fit_args, fit_kwargs, *fit_arg_names[:2])
```

Comment:
This change broke `test_autolog_emits_warning_message_when_model_prediction_fails` which assumes `_logger.warning` is called on `predict` failure here.